### PR TITLE
Add Actions for tests & pre-commit

### PR DESCRIPTION
**Pull request by simonbohnen**

### CODE REVIEW

**Comment by simonbohnen** [comment]
@janjagusch Is there a way to test out the actions in this PR?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - main
+
+jobs:
+  run_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install mamba & create environment
+        uses: mamba-org/provision-with-micromamba@main
```
Comment (janjagusch) [inline .github/workflows/ci.yml]: I suggest using a tag here.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install mamba & create environment
+        uses: mamba-org/provision-with-micromamba@v15
+
+      - name: Install quetz-client
+        shell: bash -l {0}
```
Comment (janjagusch) [inline .github/workflows/ci.yml]: I suggest setting the shell globally.,

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    shell: bash -l {0}
+
+jobs:
+  run_tests:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install mamba & create environment
```
Comment (janjagusch) [inline .github/workflows/ci.yml:22]: Can you also add the pre-commit checks, please?

Comment (simonbohnen) [reply .github/workflows/ci.yml:22]: Yep will do 👌🏼

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+name: Run tests
```
Comment (janjagusch) [inline .github/workflows/ci.yml]:
```suggestion
name: CI
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  run_tests:
+    name: Run tests
```
Comment (janjagusch) [inline .github/workflows/ci.yml]:
```suggestion
  pytest:
    name: Run pytest
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Run pytest
+        uses: pavelzw/pytest-action@v2
+        with:
+          verbose: true
+          emoji: true
+          job-summary: true
+          click-to-expand: true
+
+  pre-commit-checks:
+    name: Run pre-commit checks
```
Comment (janjagusch) [inline .github/workflows/ci.yml]:
```suggestion
  pre-commit:
    name: Run pre-commit
```

**Review by janjagusch — COMMENTED** [review]
Nie! Just some more pedantic naming stuff from my side.

**Review by janjagusch — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  pre-commit:
+    name: Run pre-commit
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3.3.0
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (janjagusch) [inline .github/workflows/ci.yml]: Can you use the same checkout step in both jobs, please?

**Review by janjagusch — DISMISSED** [review]

**Review by janjagusch — APPROVED** [review]

**Pull request merged**